src/WebCompiler/Config/Config.cs

```diff
@@ -82,7 +83,42 @@ internal bool CompilationRequired()
             if (!output.Exists)
                 return true;
 
-            return input.LastWriteTimeUtc > output.LastWriteTimeUtc;
+            if (input.LastWriteTimeUtc > output.LastWriteTimeUtc)
+                return true;
+
+            return HasDependenciesNewerThanOutput(input, output);
+        }
+
+        private bool HasDependenciesNewerThanOutput(FileInfo input, FileInfo output)
+        {
+            var projectRoot = new FileInfo(FileName).DirectoryName;
+            var dependencies = DependencyService.GetDependencies(projectRoot, input.FullName);
+
+            if (dependencies != null)
+            {
+                string key = input.FullName.ToLowerInvariant();
+                return CheckForNewerDependenciesRecursively(key, dependencies, output);
+            }
+
+            return false;
+        }
+
+        private bool CheckForNewerDependenciesRecursively(string key, Dictionary<string, Dependencies> dependencies, FileInfo output)
+        {
+            if (!dependencies.ContainsKey(key))
+                return false;
+
+            foreach (var file in dependencies[key].DependentOn.ToArray())
+            {
+                var fileInfo = new FileInfo(file);
+                if (fileInfo.LastWriteTimeUtc > output.LastWriteTimeUtc)
```

Comment:
According to the documentation if the file doesn't exist it will return `12:00 midnight, January 1, 1601 A.D. (C.E.) Coordinated Universal Time (UTC)`, but it would of course be good practice to just check it in any case.

I'll add the check.